### DISCUSSION TITLE
palindrome problem

### DISCUSSION THREAD

**Discussion by demonic-dev** [salva-imm/algo-and-bs-practice-rs#15] [Q&A] [answered]
why did you create fixed size array for palindrome? isn't it supposed to be O(1) space?

**Comment by salva-imm** [comment, marked as answer]
actually it is : ))) 
I'm not allocating extra space . but it's not a general solution
in real-world problems, I can not do that, because `N` is not limited to `100,000` or something like this 
you're right, first I implemented this by fast/slow pointers, and it was slow: ) I tried to simply optimize it.
